src/pheromones/hivemind-host/include/hivemind-host/NetworkApiDTO.h

```diff
@@ -0,0 +1,47 @@
+#ifndef PROPOLIS_NETWORKAPIDTO_H
+#define PROPOLIS_NETWORKAPIDTO_H
+
+#include "IpDiscoveryDTO.h"
+#include <message.pb.h>
+#include <variant>
+
+/**
+ * @brief Typedef to reuse for each possible type of the DTO
+ */
+using NetworkApiDTOType = std::variant<std::monostate, IPDiscoveryDTO>;
+
+class NetworkApiDTO {
+  public:
+    NetworkApiDTO(const NetworkAPI& networkApi);
+
+    NetworkApiDTO(const IPDiscoveryDTO& ipDiscovery);
+
+    /**
+    *@brief default constructor, the network api call will be initialized as a monostate (ie, no ip)
+    */
+    NetworkApiDTO();
```

Comment:
Again, why a default constructor, we want to avoid initialization in invalid state

Reply:
Used in NetworkApiHandler in HiveConnect